ir/src/constraints/mod.rs

```diff
@@ -264,6 +266,11 @@ impl Constraints {
                     domain,
                 )?;
 
+                let contains_ranom_value = self.contains_rand_value(&rhs.root_idx());
+                if contains_ranom_value && lhs.trace_segment() == DEFAULT_SEGMENT {
+                    return Err(SemanticError::InvalidUsage("Random Values cannot be used in boundary constraints defined against main trace columns".to_string()));
+                }
+
```

Comment:
Let's remove this for now and deal with it in a future PR. Then the `DEFAULT_SEGMENT` doesn't need to be `pub` anymore